pkg/collector/collector.go

```diff
@@ -174,7 +174,7 @@ func ParseHPAMetrics(hpa *autoscalingv2.HorizontalPodAutoscaler) ([]*MetricConfi
 
 	// TODO: validate that the specified metric names are defined
 	// in the HPA
-	var parser annotations.AnnotationConfigMap
+	parser := annotations.NewAnnotationConfigMap()
```

Comment:
Maybe just `parser := make(annotations.AnnotationConfigMap)` instead of a dedicated function only calling make? :)